disk-utils/fdisk.c

```diff
@@ -919,8 +919,10 @@ int print_partition_info(struct fdisk_context *cxt)
 		return rc;
 	}
 
-	if ((rc = fdisk_label_get_fields_ids_all(lb, cxt, &fields, &nfields)))
-		goto clean_data;
+	if ((rc = fdisk_label_get_fields_ids_all(lb, cxt, &fields, &nfields))) {
+    	fdisk_unref_partition(pa);	
+		return rc;
+	}
```

Comment:
Unrelated change? Please split it into its own commit with a descriptive message.